src/main/scala/com/typesafe/sbt/packager/graalvmnativeimage/GraalVMNativeImagePlugin.scala

```diff
@@ -166,35 +186,62 @@ object GraalVMNativeImagePlugin extends AutoPlugin {
     * This can be used to build a custom build image starting from a custom base image. Can be used like so:
     *
     * ```
-    * (containerBuildImage in GraalVMNativeImage) := generateContainerBuildImage("my-docker-hub-username/my-graalvm").value
+    * (containerBuildImage in GraalVMNativeImage) := generateContainerBuildImage("my-docker-hub-username/my-graalvm", Some("arm64")).value
     * ```
     *
     * The passed in docker image must have GraalVM installed and on the PATH, including the gu utility.
     */
-  def generateContainerBuildImage(baseImage: String): Def.Initialize[Task[Option[String]]] =
+  def generateContainerBuildImage(
+    baseImage: String,
+    platformArch: Option[String] = None
+  ): Def.Initialize[Task[Option[String]]] =
     Def.task {
+      import sys.process._
+
       val dockerCommand = (DockerPlugin.autoImport.dockerExecCommand in GraalVMNativeImage).value
       val streams = Keys.streams.value
+      val hostPlatform =
+        (dockerCommand ++ Seq("system", "info", "--format", "{{.OSType}}/{{.Architecture}}")).!!.trim
+      val platformValue = platformArch.getOrElse(hostPlatform)
 
       val (baseName, tag) = baseImage.split(":", 2) match {
         case Array(n, t) => (n, t)
         case Array(n)    => (n, "latest")
       }
 
       val imageName = s"${baseName.replace('/', '-')}-native-image:$tag"
+
       import sys.process._
-      if ((dockerCommand ++ Seq("image", "ls", imageName, "--quiet")).!!.trim.isEmpty) {
+      val buildContainerExists = (dockerCommand ++ Seq(
+        "image",
+        "ls",
+        "--filter",
+        s"label=com.typesafe.sbt.packager.graalvmnativeimage.platform=$platformValue",
+        "--quiet",
+        imageName
+      )).!!.trim.nonEmpty
+      if (!buildContainerExists) {
         streams.log.info(s"Generating new GraalVM native-image image based on $baseImage: $imageName")
 
         val dockerContent = Dockerfile(
           Cmd("FROM", baseImage),
           Cmd("WORKDIR", "/opt/graalvm"),
           ExecCmd("RUN", "gu", "install", "native-image"),
           ExecCmd("RUN", "sh", "-c", "ln -s /opt/graalvm-ce-*/bin/native-image /usr/local/bin/native-image"),
-          ExecCmd("ENTRYPOINT", "native-image")
+          ExecCmd("CMD", "native-image")
```

Comment:
I will never get the difference in my head. Why is this change from entrypoint to cmd needed?